rust/template/differential_datalog/program/mod.rs

```diff
@@ -640,25 +637,22 @@ pub struct RunningProgram {
     /// Producer sides of channels used to send commands to workers.
     /// We use async channels to avoid deadlocks when workers are blocked
     /// in `step_or_park`.
-    senders: Vec<mpsc::Sender<Msg>>,
+    senders: Vec<Sender<Msg>>,
     /// Channels to receive replies from worker threads. We could use a single
     /// channel with multiple senders, but use many channels instead to avoid
     /// deadlocks when one of the workers has died, but `recv` blocks instead
     /// of failing, since the channel is still considered alive.
-    reply_recv: Vec<mpsc::Receiver<Reply>>,
+    reply_recv: Vec<Receiver<Reply>>,
     relations: FnvHashMap<RelId, RelationInstance>,
-    /// Join handle of the thread running timely computation.
-    thread_handle: Option<thread::JoinHandle<Result<(), String>>>,
-    /// Timely worker threads.
-    worker_threads: Vec<thread::Thread>,
+    worker_guards: Option<WorkerGuards<Result<(), String>>>,
```

Comment:
Looks like this causes a couple of problems.  First, HDDlog is no longer `Send`, which breaks the server_api test: https://gitlab.com/ddlog/differential-datalog-kixiron/-/jobs/887145084
(This has been fixed in the latest timely: https://github.com/TimelyDataflow/timely-dataflow/issues/296, but we are still using an older version from crates.io).

Second, removing the timely thread seems to have altered the error handling path causing the nested panic error in another test: https://gitlab.com/ddlog/differential-datalog-kixiron/-/jobs/887145073

I need more time to investigate these issues, i.e., find a workaround for the former and debug the latter.

Reply:
The former can be fixed with an `unsafe impl Send for HDDlog {}` if you're comfortable with it, the latter is more tricky though

Reply:
- I think `unsafe impl` is ok.
- I haven't looked closely into the second issue yet. I suspect it's just a matter of more careful error handling now that we don't have a separate thread to catch panic's.